src/library/scala/util/control/Exception.scala

```diff
@@ -13,24 +13,125 @@ package control
 import scala.reflect.{ ClassTag, classTag }
 import scala.language.implicitConversions
 
-
 /** Classes representing the components of exception handling.
- *  Each class is independently composable.  Some example usages:
+ *
+ *  Each class is independently composable.
+ *
+ *  This class differs from [[scala.util.Try]] in that it focuses on composing exception handlers rather than
+ *  composing behavior.  All behavior should be composed first and fed to a [[Catch]] object using one of the
+ *  `opt`, `either` or `withTry` methods. Taken together the classes provide a DSL for composing catch and finally
+ *  behaviours.
+ *
+ *  === Examples ===
+ *
+ *  Create a `Catch` which handles specified exceptions.
  *  {{{
  *  import scala.util.control.Exception._
  *  import java.net._
  *
  *  val s = "http://www.scala-lang.org/"
+ *
+ *  // Option[URL]
```

Comment:
This is the raw REPL output from evaluating the first example expression by expression,

```
scala> import scala.util.control.Exception._
import scala.util.control.Exception._

scala> import java.net._
import java.net._

scala> val s = "http://www.scala-lang.org/"
s: String = http://www.scala-lang.org/

scala> val x1 = catching(classOf[MalformedURLException]) opt new URL(s)
x1: Option[java.net.URL] = Some(http://www.scala-lang.org/)

scala> val x2 = catching(classOf[MalformedURLException], classOf[NullPointerException]) either new URL(s)
x2: scala.util.Either[Throwable,java.net.URL] = Right(http://www.scala-lang.org/)

scala> val x3 = catching(classOf[MalformedURLException], classOf[NullPointerException]) withTry new URL(s)
x3: scala.util.Try[java.net.URL] = Success(http://www.scala-lang.org/)

scala> val defaultUrl = new URL("http://example.com")
defaultUrl: java.net.URL = http://example.com

scala> val x4 = failAsValue(classOf[MalformedURLException])(defaultUrl)(new URL("htt/xx"))
x4: java.net.URL = http://example.com
```